### ISSUE TITLE
Failed to handle incoming request - finish_link

### ISSUE THREAD

**Issue by halkeye** [tulir/mautrix-signal#81]
I'm trying to re-link my user after my app was deleted from my phone

mautrix-signal:
```
[2021-04-05 00:04:14,528] [ERROR@mausignald] Failed to handle incoming request {"id":"293bdf37-a98c-445c-ab39-c00a0a48d997","type":"finish_link","error":{},"error_type":"RequestProcessingError"}
Traceback (most recent call last):
  File "/opt/mautrix-signal/mausignald/errors.py", line 83, in make_response_error
    error_class = response_error_types[error_type]
KeyError: 'RequestProcessingError'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/mautrix-signal/mausignald/rpc.py", line 160, in _read_loop
    await self._handle_incoming_line(line_str)
  File "/opt/mautrix-signal/mausignald/rpc.py", line 141, in _handle_incoming_line
    self._run_response_handlers(UUID(req_id), req_type, req)
  File "/opt/mautrix-signal/mausignald/rpc.py", line 119, in _run_response_handlers
    waiter.set_exception(make_response_error(req))
  File "/opt/mautrix-signal/mausignald/errors.py", line 85, in make_response_error
    return ResponseError(data, error_type=error_type)
  File "/opt/mautrix-signal/mausignald/errors.py", line 32, in __init__
    msg = message_override or data["message"]
KeyError: 'message' 
```

signald:
```
00:02:55.108 [main] INFO  io.finn.signald.db.AccountsTable - migrating account if needed: **********34/**********************************d8
00:02:55.114 [main] INFO  io.finn.signald.Main - Binding to socket /signald/signald.sock
00:02:55.134 [main] INFO  io.finn.signald.Main - Started signald 0.13.0
00:02:59.790 [socketlistener] INFO  io.finn.signald.SocketHandler - Client connected
00:04:14.528 [socketlistener] ERROR finish_link - error while handling request
org.asamk.signal.UserAlreadyExists: null
        at io.finn.signald.ProvisioningManager.finishDeviceLink(ProvisioningManager.java:88) ~[signald.jar:unspecified]
        at io.finn.signald.clientprotocol.v1.FinishLinkRequest.run(FinishLinkRequest.java:48) ~[signald.jar:unspecified]
        at io.finn.signald.clientprotocol.v1.FinishLinkRequest.run(FinishLinkRequest.java:36) ~[signald.jar:unspecified]
        at io.finn.signald.clientprotocol.Request.<init>(Request.java:153) [signald.jar:unspecified]
        at io.finn.signald.SocketHandler.run(SocketHandler.java:143) [signald.jar:unspecified]
        at java.lang.Thread.run(Unknown Source) [?:?]                                                                                                                                                                                                                                                                                                      ```

**Comment by podiki** [comment]
Looks similar to #69 In brief, the workaround was to remove signald and link from scratch. Maybe this is an indication this needs to be handled properly though.

**Comment by halkeye** [comment]
ah sorry my bad, i should have searched first. I just wanted to record the details before I wiped my install.

**Comment by podiki** [comment]
No worries. This confirms it is a wider issue perhaps.

**Comment by halkeye** [comment]
For my specific issue, I did try out signaldctl and sqlite3, I don't have the windows open, but there was a row in accounts db for my user with registration = false. Deleting the row (tried once via sqlite3 and once via signalctl) didn't fix the issue. I eventually deleted the whole install and data for both mautrix-signal and signald and it did fix it, but I suspect upstream signald has some cache or something.
